tests/5.0/parallel_reduction/test_parallel_reduction_multiple_array_elements.c

```diff
@@ -0,0 +1,38 @@
+//===--- test_parallel_reduction_nested.c -----------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// This is a simple test of the parallel directive with a reduction clause
+// which performs a reduction on two different elements of the same array,
+// in an offloaded region.
+//
+////===----------------------------------------------------------------------===//
+#include <assert.h>
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+static double temps[2];
+
+int main() {
+  int errors = 0;
+
+  OMPVV_TEST_OFFLOADING;
+
+  temps[0] = 0;
+  temps[1] = 0;
+
+#pragma omp target map(tofrom: temps)
+  {
+#pragma omp parallel reduction(+:temps[0], temps[1])
+    {
+      temps[0] += 1;
+      temps[1] += 1;
+    }
+  }
+
+  OMPVV_TEST_AND_SET_VERBOSE(errors, temps[0] != 1 || temps[1] != 1);
```

Comment:
* If I use `reductions(+:temps)` as proposed variant in Issue 151 (i.e. #151), I get with GCC 12 and LLVM/Clang 11 and host fallback temps[0] == temps[1] == 12. — Thus, the assumption that the parallel region has only one thread seems to be far fetched.

* I think it makes sense to add this full-array variant as Issue 151 states that a compiler gave `nan` in this case, which indicates a worthwhile check. I do note that the OpenMP spec (here: 5.2) states under "5.5.5 Properties Common to All Reduction Clauses":

"If a list item is an array or array section, it will be treated as if a reduction clause would be applied to each separate element of the array section."

Thus, using `reduction(+:temps)` should be valid.

* I have not tried to understand whether `reduction(+:temps[0], temps[1])` is valid or not. However, the restriction "Any array element must be specified at most once in all list items on a directive." implies that it is valid. (From "5.5.5 Properties Common to All Reduction Clauses" in OpenMP)

Reply:
> * I have not tried to understand whether `reduction(+:temps[0], temps[1])` is valid or not. However, the restriction "Any array element must be specified at most once in all list items on a directive." implies that it is valid. (From "5.5.5 Properties Common to All Reduction Clauses" in OpenMP)

Jakub believes:
> No, IMNSHO this is invalid.
> `reduction` is a privatization clause, and one can't privatize two elements of  the same variable separately.  It is valid to do `#pragma omp parallel reduction(+:temps[:2])`, but not `reduction(+:temps[0], temps[1])`

Reply:
Regarding Jakub's comments (cf. OpenMP 5.2), `reduction` belongs to "Properties: privatization" and in the glossary item "private variable" reads "... A variable that is part of another variable (as an array element or a structure element) cannot be made private independently of other components. ...".

_(Regarding the misleading spec bit, I filled the OpenMP spec issue 3187.)_